src/delete.ts

```diff
@@ -15,7 +15,13 @@ export function getVersionIds(input: Input): Observable<string[]> {
       input.packageName,
       input.numOldVersionsToDelete,
       input.token
-    ).pipe(map(versionInfo => versionInfo.map(info => info.id)))
+    ).pipe(
```

Comment:
@Hsn723 did you look into a solution that exclude the versions from the query in the first place? If that is even possible.

Could you run into a situation where there'll never be any versions left. The case where `numOldVersionsToDelete` is smaller than the number of versions matching you regular expression? Would one have to "know your versions" and set this value higher to avoid that or what do you think?

Kind of related: Do you know if this action could remove *all* versions or will it at least leave one?

Reply:
I did look into passing the filter into the query however the current GitHub GraphQL API v4 does not support any filtering at this time.

> Could you run into a situation where there'll never be any versions left.

Indeed, there's the possibility of the oldest `numOldVersionsToDelete` versions all match the filtering pattern and therefore ending up with nothing to delete. Unfortunately I don't have an elegant solution other than "know your versions" as you mentioned. Or, having a versioning scheme that allows some predictability. Automatically counting up is also an option, however that will require several passes.

> Do you know if this action could remove all versions or will it at least leave one?

Yes, even with the initial implementation this action can very well remove all versions, for instance if `numOldVersionsToDelete` is set higher to the total number of packages. Leaving at least one (the latest version) would indeed be desirable, although probably out of the scope of the current PR (I can gladly prepare another PR for that though).

Reply:
Thanks for your valuable replies. I can see the complexities.

I find it odd that the default implementation would let you delete all versions. I'm afraid to use the action without this "protection".

Also, the `numOldVersionsToDelete` is kind of up-side-down isn't it? Wouldn't it make more sense with a `keepNumVersions` + what your PR providers?

> Leaving at least one (the latest version) would indeed be desirable, although probably out of the scope of the current PR

Yeah, that would be a different PR.

Reply:
`keepNumVersions` has also been proposed in #5, also that one is also tricky with the current GitHub GraphQL API v4 as the API doesn't provide a way to return "all" package versions unless you call the API asking for the "first N packages" with a large enough number.

By the way, if #5 gets implemented it would also be possible to combine it with the `ignore-package-names` proposed in this PR.

Reply:
> API doesn't provide a way to return "all" package versions unless you call the API asking for the "first N packages" with a large enough number.

I guess any, not necessarily large, number would suffice. As long as you don't add more than whatever that number is. At each delete action execution it would carve away old versions until there are no more of them. Isn't that how it would/could work?

> By the way, if #5 gets implemented it would also be possible to combine it with the ignore-package-names proposed in this PR.

Good to know, thank you.

Reply:
> At each delete action execution it would carve away old versions until there are no more of them. 

In theory yes, for instance if you set `keepNumVersions` to 5 and call the API every time requesting the 10 latest versions. What could possibly betray user expectations however is that this way you would end up deleting old (but not oldest) packages and slowly working your way down when the user would rather have oldest packages deleted first. Though the reality is that automation with the current API has no "one size fits all" and some users' expectations may be betrayed regardless.

Reply:
I see. Thanks for the explanation.